analyzers/src/SonarAnalyzer.CSharp/Rules/MethodParameterUnused.cs

```diff
@@ -69,25 +69,26 @@ private static MethodContext CreateContext(SonarSyntaxNodeReportingContext c)
             {
                 return new MethodContext(c, method);
             }
-            else if (c.Node.Kind() == SyntaxKindEx.LocalFunctionStatement)
+            else if (c.Node.IsKind(SyntaxKindEx.LocalFunctionStatement))
             {
-                return new MethodContext(c, (LocalFunctionStatementSyntaxWrapper)c.Node);
+                var localFunction = (LocalFunctionStatementSyntaxWrapper)c.Node;
+                return new MethodContext(c, localFunction);
```

Comment:
You can fully revert the line. The rename isn't needed
```suggestion
                return new MethodContext(c, (LocalFunctionStatementSyntaxWrapper)c.Node);
```